build-system/tasks/performance/download-images.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const {CONTROL, maybeCopyImageToCache, urlToCachePath} = require('./helpers');
+const {JSDOM} = require('jsdom');
+
+/**
+ * Lookup URL from cache. Look at tags that could use images. Download images
+ * with a local src to cache.
+ *
+ * @param {string} url
+ */
+function downloadImagesFromTags(url) {
```

Comment:
nit: in its current form, this is less of a "download" and more of a "copy" right?